static/js/src/advantage/distributor/utils/FormContext.tsx

```diff
@@ -131,95 +130,68 @@ export const FormProvider = ({
   const updatedChannelProductList = useMemo(() => {
     const rawChannelProductListings = window.channelProductList;
     const offerItems = offer?.items || [];
+    const offerQuarter = offer?.exclusion_group || "";
     const updatedChannelProductList: ProductListings = {};
 
-    const extractNameAndVersion = (listing: any) => {
-      const nameWithVersion = listing?.name || ""; // ex: "uai-essential-desktop-1y-channel-eur-v2"
-      const nameWithoutVersion =
-        listing?.name?.lastIndexOf("-") === -1
-          ? nameWithVersion
-          : nameWithVersion.substring(0, listing?.name?.lastIndexOf("-")); // ex: "uai-essential-desktop-1y-channel-eur"
-      const version =
-        listing?.metadata?.find((data: Metadata) => data.key === "version")
-          ?.value || "1"; // "2"
-      return { nameWithoutVersion, version }; // ex: { nameWithoutVersion: "uai-essential-desktop-1y-channel-eur", version: "2" }
+    const extractNameWithoutQuarter = (listing: any) => {
```

Comment:
(An easy fix for this would be to just manually construct the keys for `updatedChannelProductList` based on the attributes of the listing and remove `extractNameWithoutQuarter`.)

Reply:
Hi @amethystant, thanks for your feedback. I thought the name would always follow the pattern (${product?.id}-${duration}y-channel-${currency}), but only the exclusion_group pattern would change.
I didn’t know the product listing included currency and duration other than the name field. I checked the raw product listing, and it includes `effectiveDays` and `currency`. I'll make some changes soon so the product listing name won't matter anymore. Thank you!